packages/session-replay-browser/src/session-replay.ts

```diff
@@ -209,44 +211,35 @@ export class SessionReplay implements AmplitudeSessionReplay {
     });
   };
 
-  evaluateTargetingAndRecord = async (targetingParams?: Pick<TargetingParameters, 'event' | 'userProperties'>) => {
+  evaluateTargetingAndCapture = async (targetingParams?: Pick<TargetingParameters, 'event' | 'userProperties'>) => {
     if (!this.identifiers || !this.identifiers.sessionId || !this.config) {
-      if (!this.identifiers?.sessionId) {
-        this.loggerProvider.warn('Session ID has not been set, cannot evaluate targeting for Session Replay.');
+      if (this.identifiers && !this.identifiers.sessionId) {
+        this.loggerProvider.log('Session ID has not been set yet, cannot evaluate targeting for Session Replay.');
       } else {
         this.loggerProvider.warn('Session replay init has not been called, cannot evaluate targeting.');
       }
-      return false;
-    }
-    // Return early if a targeting match has already been made
-    if (this.sessionTargetingMatch) {
-      if (!this.recordCancelCallback) {
-        this.loggerProvider.log('Session replay capture beginning due to targeting match.');
-        this.recordEvents();
-      }
-      return this.sessionTargetingMatch;
-    }
-
-    let eventForTargeting = targetingParams?.event;
-    if (
-      eventForTargeting &&
-      Object.values(SpecialEventType).includes(eventForTargeting.event_type as SpecialEventType)
-    ) {
-      eventForTargeting = undefined;
+      return;
     }
 
-    this.sessionTargetingMatch = await evaluateTargetingAndStore({
-      sessionId: this.identifiers.sessionId,
-      config: this.config,
-      targetingParams: { userProperties: targetingParams?.userProperties, event: eventForTargeting },
-    });
+    if (this.config.targetingConfig && !this.sessionTargetingMatch) {
+      let eventForTargeting = targetingParams?.event;
+      if (
+        eventForTargeting &&
+        Object.values(SpecialEventType).includes(eventForTargeting.event_type as SpecialEventType)
+      ) {
+        eventForTargeting = undefined;
+      }
 
-    if (this.sessionTargetingMatch) {
-      this.loggerProvider.log('Session replay capture beginning due to targeting match.');
-      this.recordEvents();
+      this.sessionTargetingMatch = await evaluateTargetingAndStore({
+        sessionId: this.identifiers.sessionId,
+        targetingConfig: this.config.targetingConfig,
+        loggerProvider: this.loggerProvider,
+        apiKey: this.config.apiKey,
+        targetingParams: { userProperties: targetingParams?.userProperties, event: eventForTargeting },
+      });
     }
 
-    return this.sessionTargetingMatch;
+    this.captureEventsIfShould();
```

Comment:
I wonder if there's a longer term way to refactor this so that the methods aren't co-dependent on knowing what each other is doing. Maybe more specifically this knows that captureEventsIfShould will use sessionTargetingMatch.

Reply:
Yeah I agree this is awkward. I wrote it this way to have a single source of truth for making the decision about capturing, the `getShouldCapture` fn. This is used for both the decision to start the rrweb record method, and to determine whether replay event properties are returned. My worry with passing around these state variables is that we may lose other important context - for example, `getShouldCapture` also takes into account the `optOut` and `captureEnabled` configs, to make a wholistic decision about replay capture. I'd love ideas about how to make the flow of logic here both bulletproof and easier to reason about though!